src/main/java/org/owasp/esapi/SecurityConfiguration.java

```diff
@@ -640,6 +640,12 @@
      */
     InputStream getResourceStream( String filename ) throws IOException;
 
+    /**
+     * Used to load antisamy-esapi.xml from a variety of different classpath locations.
+     *
+     * @param fileName The resource file filename.
+     */
+	InputStream getResourceStreamFromClasspath( String fileName );
```

Comment:
This API makes me a little nervous.  It introduces a high potential for a resource leak, exclusively dependent on which concrete implementation is used by the client/caller.

In short unless every single use of this API is wrapped in either a Try with resources or a try/catch block with appropriate close, we're introducing a long-term maintenance risk that may later explode.

In the past I've tried to remove public API's that require the client/call of the API to have implicit handling knowledge and tried to make it so that my implementations control how the resources are managed.  

Documentation that state "The client is responsible for the cleanup of the stream" are always missed during the implementations.

If it were possible to use lambdas here, we could pass in a function that could ingest the stream, rather than returning the stream.  That would require full Java8 compliance, and I'm not sure the project is there yet.

Alternatively, constructing a callback API to allow that interaction would be another way to account for it.  IE
```
public interface StreamIngest {
  void ingest (InputStream);
}

//SecurityConfiguration
void readClasspathResource(String filename, StreamInjest streamIngest)

//DefaultSecurityConfiguration
void readClasspathResource(String filename, StringInjest streamIngest) {
  try (InputStream stream = privateMethodToGetStreamFromClasspath(filename)) {
     streamIngest.ingest(stream);  
  }
}
```

Reply:
please keep in mind that InputStream getResourceStream( String fileName ); already exists as a public method. How much different is the new method from this?

Reply:
@jeremiahjstacey  - I think we should really be concerned about leaks that we might cause in ESAPI directly. While it would be nice to do things that make it difficult for a client calling ESAPI to screw up and not shoot themselves in the foot, that's not always possible. (But then after spending 20 years developing in C++, that's probably understandable. ;-) I general though, we can't be overly concerned about this and I think that the point raised by @HJW8472 is legitimate, even if we were much more lax when that was written so this is closer to legacy code in that sense. I think we can avoid much of this by just making this a private static method in HTMLValidationRule and then bullet proof it at that point.

Reply:
This particular concern was addressed by leaving DefaultSecurityConfiguration unaltered (except for a new property name) and moving this to a private method in HTMLValidationRule as discussed above.